hooks/scheduler.js

```diff
@@ -49,7 +51,7 @@ class Scheduler {
     this.stopping = false;
 
     // Create a promise that we're done looping
-    this.done = poll().catch((err) => {
+    this.done = this.loopUntilStopped().catch((err) => {
       debug("Error: %s, as JSON: %j", err, err, err.stack);
       throw err;
```

Comment:
This needs to `this.emit('error', err);` and we have to subclass eventEmitter, `class Scheduler extends events.EventEmitter`.